flowpipe/node.py

```diff
@@ -160,6 +161,63 @@ def evaluate(self):
 
         return outputs
 
+    def _evaluate(self, **inputs):
+        # Look if iteration is needed
+        # Iterables is a dictionary with each entry an list of inputs to iterate over later-on
+        is_iteration_detected, iterables = False, {}
+        # If itertation is disabled just continue normally
+        if self.iteration_count != -1:
+            # Iterate over the inputs
+            for key, value in inputs.items():
+                # Input value is an IterationPlug so iteration is needed
+                if isinstance(value, IterationPlug):  
+                    is_iteration_detected = True
+                    iterables[key] = [{key: x} for x in value]  # Create a list of kwargs to iterate over lateron.
+                    if self.iteration_count is not None:
+                        # Check the number of iterations if provided
+                        assert len(value) == self.iteration_count, f"Length ({len(value)}) of iterable `{key}` is unequal to expected length {self.iteration_count}"
+                # Special case for subinputs
+                elif isinstance(value, dict):
+                    for subkey, subvalue in value.items():
+                        if isinstance(subvalue, IterationPlug):  # Same as above but a layer deeper.
+                            is_iteration_detected = True
+                            if key not in iterables:
+                                iterables[key] = [{key: {subkey: x}} for x in subvalue]
+                            else:
+                                iterables[key] = [{key: {**x[key], subkey: y}} for x, y in zip(iterables[key], subvalue)]
+                            if self.iteration_count is not None:
+                                assert len(subvalue) == self.iteration_count, f"Length ({len(subvalue)}) of iterable `{key}` is unequal to expected length {self.iteration_count}"
+
+        
+        if is_iteration_detected: # Iterate if iteration is detected
+            outputs = {}
+            for arguments in zip(*iterables.values()):  # Loop over the list of kwargs in iterables
+                [inputs.update(arg) for arg in arguments]  # Update the inputs
```

Comment:
I'm not a fan of using a list comprehension with a function that has side-effects. Also, I don't understand what structure inputs has after this. Isn't it easier to
```
for arguments in zip(*interables.values()):
    for arg in arguments:
        iter_args = inputs.copy()
        iter_args.update(arg)
        output = ...
```